### PR TITLE
Fix rrpn proposal validation

### PR DESCRIPTION
**Pull request by nijkah**
https://github.com/facebookresearch/detectron2/blob/08f617cc2c9276a48e7e7dc96ae946a5df23af3f/detectron2/modeling/proposal_generator/proposal_utils.py#L104
```python
        if not valid_mask.all():
            if training:
                raise FloatingPointError(
                    "Predicted boxes or scores contain Inf/NaN. Training has diverged."
                )
            boxes = boxes[valid_mask]
            scores_per_img = scores_per_img[valid_mask]
            lvl = lvl[valid_mask]
```

Unlike function **find_top_rpn_proposals**, **find_top_rrpn_proposals** does not include valid masking for level indices.
When the program goes into if condition, it gives an undesirable error says that "shape does not match".

I add a handling code same with **find_top_rpn_proposals**.

### CODE REVIEW

**Review by ppwwyyxx — APPROVED** [review]

**Comment by nijkah** [comment]
@ppwwyyxx 
Is there something that I have to check for merging?

I couldn't find any information for final decision of Merge Pull Request.

**Comment by ppwwyyxx** [comment]
@zhanghang1989 Could you merge this bug fix (assuming tests will pass)? Thanks!

**Pull request closed without merging**
